### PR TITLE
feat(target-postgres): recover domain enums from hash-verified membership checks (domain-enum-inference slice 2)

### PR DESCRIPTION
**Pull request by wmadden-electric**
Slice 2 of the domain-enum-inference project (spec: `projects/domain-enum-inference/slices/recover-enums-from-derived-checks/spec.md`, in this diff).

## What this does

A database that Prisma Next migrated now round-trips its domain enums. Before: `contract infer` pulled a text-backed enum column back as a plain `String` plus an opaque `@@check`. After:

```prisma
enum AccountsRole {
  user  = "user"
  admin = "admin"

  @@type("pg/text@1")
}

model Accounts {
  id   Int          @id
  role AccountsRole

  @@map("accounts")
}
```

## How recovery is proven

`contract infer` recovers a domain `enum` only from a CHECK constraint Prisma Next itself created (project spec § Path A). For each live check whose name is wire-shaped with a `<table>_<column>_check` membership prefix, inference harvests the single-quoted literals from the Postgres reprint (a text scan, never a predicate parser), re-renders the membership predicate through `postgresRenderCheckExpressions`, re-hashes, and recovers the enum only when the recomposed wire name equals the live constraint's name byte-for-byte. A proven column gets a top-level `enum` block and is typed by it; the proven constraint emits no `@@check` and no `@noCheck` because authoring re-derives it. Anything unproven is untouched — a wrong harvest can never affect constraint emission.

Recovery maps only exact codec target spellings (`text`, `varchar`, `character varying`, `char`, `character`). A parameterized column such as `varchar(20)` recovers nothing and keeps its `@@check`: `@@type` re-emits the codec's bare target type, so recovering would silently drop the length and the planner would widen the column.

Recovered-enum names uniquify against models, native enums, scalar type names and each other (numeric suffix, never a throw — project spec Locked decision 6), and the reserved scalar-name set now covers the target-contributed type names (`Uuid`, `VarChar`, `BigIntNumber`, …) instead of only the nine framework scalars.

## The round-trip proof

A new e2e journey (`infer-roundtrip-fidelity.e2e.test.ts`) drives the full loop: emit a contract with a domain enum (non-alphabetical member order) and a native enum on the same table, `db init` (the toolchain installs the derived wire-named membership check itself — no precomputed names), then re-infer. The re-pull returns the same enum under its derived name in the authored order, the recovered block top-level with the native enum inside the namespace wrap, and the re-pulled contract emits, verifies clean, and plans zero operations. Disabling recovery makes the journey fail.

## Out of scope (slice 3)

Hand-written membership checks (Path B: `@noCheck(membership)` + verbatim `@@check(map:)`) are the next slice. Known hazards deferred there are recorded in the slice plan's Open items: empty-string harvests, `E'a\b'` backslash doubling, and the Supabase generator's masked descriptor gaps.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * PostgreSQL schema inference now recovers domain-style enums from validated membership checks.
  * Recovered enums preserve member order, quoted values, list fields, and database namespaces.
  * Generated contracts avoid naming collisions and retain stable enum identifiers.
  * Round-trip workflows now re-emit recovered enums without unnecessary changes.

* **Tests**
  * Added coverage for literal extraction, enum recovery, edge cases, and end-to-end round trips.

### CODE REVIEW

File: packages/3-targets/3-targets/postgres/src/core/psl-infer/harvest-check-literals.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Extracts the single-quoted string literals from a live CHECK expression
+ * reprint, in order of appearance, with doubled quotes unescaped
+ * (`'O''Brien'` → `O'Brien`). A text scan only — no predicate shape is
+ * recognized; casts, operators, and identifiers are skipped. An expression
+ * with no literals yields an empty list.
+ */
+export function harvestCheckLiterals(expression: string): string[] {
```
Comment (wmadden) [inline packages/3-targets/3-targets/postgres/src/core/psl-infer/harvest-check-literals.ts:8]: This is incomprehensible. "Harvest" is not our ubiquitous language

File: packages/3-targets/3-targets/postgres/src/core/psl-infer/recover-domain-enums.ts
Comment (wmadden) [inline packages/3-targets/3-targets/postgres/src/core/psl-infer/recover-domain-enums.ts:1]: what does it mean to recover enums? is it different to harvesting them

**Review by wmadden — CHANGES_REQUESTED** [review]
